src/content-helper/editor-sidebar/excerpt-suggestions/class-excerpt-suggestions.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Content Helper: Excerpt Generator feature class
+ *
+ * @package Parsely
+ * @since   3.13.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Content_Helper;
+
+use Parsely\Content_Helper\Editor_Sidebar\Editor_Sidebar_Feature;
+
+/**
+ * Content Helper: Excerpt Generator feature class
+ *
+ * @since 3.13.0
+ * @since 3.17.0 Renamed to Excerpt Suggestions and converted to an Editor Sidebar feature.
+ */
+class Excerpt_Suggestions extends Editor_Sidebar_Feature {
+	/**
+	 * Returns the feature's name.
+	 *
+	 * @since 3.17.0
+	 *
+	 * @return string
+	 */
+	public static function get_feature_name(): string {
+		return 'excerpt_suggestions';
+	}
+
+	/**
+	 * Returns the feature's filter name.
+	 *
+	 * This is deprecated in favor of the new wp_parsely_current_user_can_use_pch_feature filter.
+	 *
+	 * @since 3.17.0
+	 *
+	 * @return string The filter name.
+	 */
+	public static function get_feature_filter_name(): string {
+		return self::get_global_filter_name() . '_excerpt_generator';
+	}
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.17.0
+	 *
+	 * @param Editor_Sidebar $editor_sidebar Instance of Editor_Sidebar class.
+	 */
+	public function __construct( Editor_Sidebar $editor_sidebar ) {
+		parent::__construct( $editor_sidebar );
+
+		// Check if the feature is disabled using the global and feature filters.
+		if ( has_filter( self::get_feature_filter_name() ) ) {
+			add_filter(
+				'wp_parsely_current_user_can_use_pch_feature',
+				array( $this, 'callback_disable_feature_via_filters' ),
+				10,
+				2
+			);
+		}
+	}
+
+	/**
+	 * Checks and disables the feature if it is disabled via the global or feature filters.
+	 *
+	 * This throws a deprecated notice if the feature filter name is in use.
+	 *
+	 * @since 3.17.0
+	 *
+	 * @param ?bool  $current_user_can_use_pch_feature Whether the current user can use the feature.
+	 * @param string $feature_name The feature's name.
+	 * @return ?bool  Returns false if the feature is disabled via filters.
+	 */
+	public function callback_disable_feature_via_filters(
```

Comment:
This function returns a bool flag, but it's name indicates that it disables the feature. Maybe we can find a name that's a better fit for it?

Reply:
Yeah, I struggled a little bit trying to find the most appropriate name for this callback function. I renamed it to `callback_is_feature_enabled`, since it hooks to `wp_parsely_current_user_can_use_pch_feature`. What do you think?

Reply:
Yes, `callback_is_feature_enabled` makes sense to me!